src/core/Akka.Remote.Tests/RemoteMessageLocalDeliverySpec.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.Remote.Transport;
+using Akka.TestKit;
+using Akka.TestKit.TestActors;
+using Xunit;
+
+namespace Akka.Remote.Tests
+{
+    /// <summary>
+    /// Came across some issues while debugging multi-node tests which indicated
+    /// that the <see cref="RemoteActorRefProvider"/> couldn't successfully decode full addresses
+    /// for local actors, such as "akka.trttl.gremlin.tcp://AttemptSysMsgRedeliverySpec@localhost:57512/user/echo",
+    /// into valid <see cref="LocalActorRef"/>s.
+    /// 
+    /// This spec is designed to verify that these types of paths, including ones with transport adapters at the front, 
+    /// can be successfully translated.
+    /// </summary>
+    public class RemoteMessageLocalDeliverySpec : AkkaSpec
```

Comment:
Used to simulate some of the bugs around `ActorRefProvider.ResolveActorRef` that were discovered inside `AttemptSysMsgDelivery`